rw_mutex.go

```diff
@@ -39,6 +39,43 @@ func NewRWMutex(collection *mgo.Collection, lockID, clientID string) *RWMutex {
 	}
 }
 
+func (m *RWMutex) tryToGetWriteLock() error {
+	err := m.collection.Update(bson.M{
+		"lockID":  m.lockID,
+		"readers": []string{},
+		"writer":  "",
+	}, bson.M{
+		"$set": bson.M{
+			"writer": m.clientID,
+		},
+	})
+	if err == nil {
+		return nil // we got the lock!
+	} else if err != mgo.ErrNotFound {
+		// This only works if the mgo.Session object has Safe mode enabled.
+		// Safe is the default but something for which we should maintain
+		// external documentation
```

Comment:
Thanks for the helpful comment

Reply:
That comment was just copy/pasted... I actually don't know what it refers to... :hide:

Reply:
©️ 🍝

Reply:
Looks like Safe is on by default - https://github.com/go-mgo/mgo/blob/v2/session.go#L534
So a consumer would have to explicitly have overwritten this setting.